graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/types/ID.kt

```diff
@@ -14,12 +14,16 @@
  * limitations under the License.
  */
 
-package com.expediagroup.graphql.exceptions
+package com.expediagroup.graphql.types
 
-import kotlin.reflect.KClass
+import com.fasterxml.jackson.annotation.JsonIgnore
+import com.fasterxml.jackson.annotation.JsonValue
 
 /**
- * Throws when the KClass is not one of the supported types for a GraphQLID
+ * Used to represent a GraphQL ID scalar type
+ * which must serialize/deserialize to a string value
  */
-class InvalidIdTypeException(kClass: KClass<*>) :
-    GraphQLKotlinException("${kClass.simpleName} is not a valid ID type, only Strings are accepted")
+data class ID(@get:JsonIgnore val value: String) {
```

Comment:
wondering whether we should rename this class to something explicit like `GraphQLID` (unsure about capitalization there...)

Reply:
I was debating this as well. The full package paths makes it pretty clear that it is a GraphQL ID but I was thinking since the name of the scalar in the schema is ID that it would make more sense to have it be the same here, because if we call it `GraphQLID` then this

```kotlin
data class Person(
    val id: GraphQLID,
    val name: String
)
```

becomes this 

```graphql
type Person {
    id: ID!
    name: String!
}
```

Reply:
👍 lets keep it as just `ID`